src/classes/RD2_OpportunityService.cls

```diff
@@ -0,0 +1,210 @@
+            }
+            //Confirm if the error handling is needed due to bulk operations
+            if (!badSRs.isEmpty()) {
+               ERR_Handler.Errors errors = ERR_Handler.getErrors(badSRs, badOpps);
+               ERR_Handler.processErrors(errors.errorRecords, ERR_Handler_API.Context.RD.name());
+            }
+        }
+        updatePaymentRecordOnCreatedOpp(createdOppIds, rdPaymentMethod);
+    }
+
+    /***
+    * @description Gets one or more Opportunity records for a specified set of Recurring Donations with close dates
+    * @param rdsWithCloseDates the Map of Recurring Donations with close dates
+    * @param contactMap the Map of contact with account ids.
+    * @return the list of Opportunities to be created.
+    */
+    private List<Opportunity> getInstallmentOpportunities(Map<npe03__Recurring_Donation__c, List<Date>> rdsWithCloseDates, Map<Id,Contact> contactMap) {
+
+        List<Opportunity> oppsToInsert = new List<Opportunity>();
+        npe03__Recurring_Donations_Settings__c rds = UTIL_CustomSettingsFacade.getRecurringDonationsSettings();
+
+        for (npe03__Recurring_Donation__c r : rdsWithCloseDates.keySet()) {
+
+            for(Integer i = 0;i< rdsWithCloseDates.get(r).size();i++) {
+                Opportunity opp = new Opportunity();
+                opp.Name = r.npe03__Donor_Name__c + ' ' + r.npe03__Amount__c.format() + ' ' + rdsWithCloseDates.get(r)[i].format() + ' ' + System.Label.stgNavRecurringDonations;
+
+                if ( r.npe03__Organization__c != null && String.isBlank(r.npe03__Contact__c)) {
+                    opp.AccountId = r.npe03__Organization__c;
+
+                } else if (contactMap.containsKey(r.npe03__Contact__c)) {
+                    Contact c = contactMap.get(r.npe03__Contact__c);
+                    opp.AccountId = c.AccountId;
+                    opp.Primary_Contact__c = c.id;
+                }
+                opp.CloseDate = rdsWithCloseDates.get(r)[i];
+                opp.StageName = rds.InstallmentOppStageName__c ;
+                opp.Amount = r.npe03__Amount__c;
+                opp.npe03__Recurring_Donation__c = r.Id;
+                if (r.npe03__Recurring_Donation_Campaign__c != null && rds.npe03__Add_Campaign_to_All_Opportunites__c ) {
+                    opp.CampaignId = r.npe03__Recurring_Donation_Campaign__c;
+                }
+                oppsToInsert.add(opp);
+            }
+          }
+
+        return oppsToInsert;
+
+    }
+
+     /***
+    * @description Update payment method on the payment records for the created opportunities.
+    * @param oppIds set of opportnity Ids
+    * @param rdPayment is the map of Recurrding donation Ids with Payment method.
+    */
+    private void updatePaymentRecordOnCreatedOpp(Set<Id> oppIds, Map<Id,String> rdPayment) {
+
+        List<npe01__OppPayment__c> paymentForUpdates = new List <npe01__OppPayment__c> ();
+
+        List<npe01__OppPayment__c> paymentRecForOpps = new List <npe01__OppPayment__c> ([
```

Comment:
I'm uncertain that the additional queries, lists and maps are needed here. If the method has a list of the Opportunity Id's inserted and a Map the PaymentMethod to use by RD, you can simply modify the query to the following so the RecurringDonation Id is available direct from the payment record.
```
        List<npe01__OppPayment__c> paymentRecords = [
                SELECT Id,npe01__Payment_Method__c,npe01__Payment_Date__c, npe01__Opportunity__c,
                npe01__Opportunity__r.npe03__Recurring_Donation__c
                FROM npe01__OppPayment__c
                WHERE npe01__Opportunity__c IN :oppIds
            ];
```

From there the loop to assign the payment method is just:
```
        for (npe01__OppPayment__c payment : paymentRecords) {
            payment.npe01__Payment_Method__c = rdPaymentMethods.get(payment.npe01__Opportunity__r.npe03__Recurring_Donation__c);
        }

        Database.SaveResult[] saveResults = Database.update(paymentRecords, false); 
```